### DISCUSSION TITLE
Ready for production?

### DISCUSSION THREAD

**Discussion by lferna** [quarkiverse/quarkus-cxf#322] [Q&A] [answered]
Hi,

I’d like to know if we can use the extension in production environments.

And probably with hundreds of req/sec.

Thanks,
KR

**Comment by shumonsharif** [comment, marked as answer]
Hi @lferna This is a community extension, and as such, supported by volunteers during their free time.  I don't know if you'll be able to get any direct answer for your question from the core contributors; but I believe what may possibly help you is any success stories from the rest of the community.

As far as your throughput requirements, I am unaware of any performance or load testing of the extension, and any feedback you or anyone else can provide with regard to that would be quite valuable.

**Reply by lferna** [reply]
Thanks @shumonsharif  

We probably give it a try in production with our customers.

Anyway, we only need the soap client (not the service) because the service is external.

What we had before is Apache CXF extension but with JVM mode (only Java 1.8) so we can send SOAP requests. So, is this native extension similar? Because looking at tests, I can see we have a service and a client. And if we use like CFX in JVM mode (but with your extension) we receive this error:  Failed to load steps from class io.quarkiverse.cxf.deployment.QuarkusCxfProcessor

Thanks.

**Reply by shumonsharif** [reply]
The extension supports both native and JVM mode.  Would you mind opening an issue with the full stack trace?  If at all possible, a reproducer project would be great too.

Also, please do keep us posted on your experiences in production - we are very curious to hear about how the extension is being leveraged!

**Comment by famod** [comment]
I'm actually using this extension successfully in production (client only, for two different backends and soon a third one).
We don't use native and we don't have hundreds of req/sec, though.

Prod is still on Java 11, but we moved to 17 for the current release that's being developed (will go live in January).
